packages/kit/src/core/dev/index.js

```diff
@@ -333,19 +333,21 @@ async function create_handler(vite, config, dir, cwd, manifest) {
 							vite.ssrFixStacktrace(error);
 							return error.stack;
 						},
-						handle_error: /** @param {Error & {frame?: string}} error */ (error) => {
-							vite.ssrFixStacktrace(error);
-							console.error(colors.bold().red(error.message));
-							if (error.frame) {
-								console.error(colors.gray(error.frame));
-							}
-							if (error.stack) {
-								console.error(colors.gray(error.stack));
-							}
-						},
 						hooks: {
 							getSession: hooks.getSession || (() => ({})),
 							handle: hooks.handle || (({ request, resolve }) => resolve(request)),
+							handleError:
+								hooks.handleError ||
+								(({ error }) => {
+									vite.ssrFixStacktrace(error);
```

Comment:
this should always be done as well. we don't want this to get dropped if the user provides an implementation

Reply:
Also check here, if the solution I created matches what you were thinking